timely_beliefs/beliefs/classes.py

```diff
@@ -395,14 +395,20 @@ def search_session(  # noqa: C901  # todo: remove after removing deprecated argu
             )
 
         # Query sensor for relevant timing properties
-        event_resolution, knowledge_horizon_fnc, knowledge_horizon_par = (
-            session.query(
-                sensor.__class__.event_resolution,
-                sensor.__class__.knowledge_horizon_fnc,
-                sensor.__class__.knowledge_horizon_par,
+        if isinstance(sensor, int):
+            sensor = (
+                session.query(sensor.__class__)
+                .filter(sensor.__class__.id == sensor)
+                .one_or_none()
             )
-            .filter(sensor.__class__.id == sensor.id)
-            .one_or_none()
+            if sensor is None:
+                raise ValueError("No such sensor")
+        elif not isinstance(sensor, SensorDBMixin):
+            raise ValueError(f"sensor {sensor} is not a {SensorDBMixin}")
+        event_resolution, knowledge_horizon_fnc, knowledge_horizon_par = (
```

Comment:
I believe we use all three of these variables only once and shortly after this line, so I find it better to just prepend "sensor." three times, instead of adding these 5 extra lines.